src/CoreWCF.Primitives/src/CoreWCF/Security/CryptoHelper.cs

```diff
@@ -0,0 +1,360 @@
+using System.Collections;
+using System.Collections.Generic;
+using CoreWCF.Channels;
+using CoreWCF;
+using System.Reflection;
+using System.Threading;
+using System.IO;
+
+using System.Runtime.InteropServices;
+using CoreWCF.IdentityModel.Tokens;
+using System.Text;
+using System.Xml;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography;
+
+using Psha1DerivedKeyGenerator = CoreWCF.IdentityModel.Psha1DerivedKeyGenerator;
+using CryptoAlgorithms = CoreWCF.IdentityModel.CryptoHelper;
+using System;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Security
+{
+
+
+    static class CryptoHelper
+    {
+        static byte[] emptyBuffer;
+        static readonly RandomNumberGenerator random = new RNGCryptoServiceProvider();
+
+        enum CryptoAlgorithmType
+        {
+            Unknown,
+            Symmetric,
+            Asymmetric
+        }
+
+        internal static byte[] EmptyBuffer
+        {
+            get
+            {
+                if (emptyBuffer == null)
+                {
+                    byte[] tmp = new byte[0];
+                    emptyBuffer = tmp;
+                }
+                return emptyBuffer;
+            }
+        }
+
+        internal static HashAlgorithm NewSha1HashAlgorithm()
+        {
+            return CryptoHelper.CreateHashAlgorithm(SecurityAlgorithms.Sha1Digest);
+        }
+
+        internal static HashAlgorithm NewSha256HashAlgorithm()
+        {
+            return CryptoHelper.CreateHashAlgorithm(SecurityAlgorithms.Sha256Digest);
+        }
+
+        [SuppressMessage("Microsoft.Security.Cryptography", "CA5354:DoNotUseSHA1", Justification = "Cannot change. Required as SOAP spec requires supporting SHA1.")]
+        internal static HashAlgorithm CreateHashAlgorithm(string digestMethod)
+        {
+            object algorithmObject = CryptoAlgorithms.GetAlgorithmFromConfig(digestMethod);
+            if (algorithmObject != null)
+            {
+                HashAlgorithm hashAlgorithm = algorithmObject as HashAlgorithm;
+                if (hashAlgorithm != null)
+                    return hashAlgorithm;
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new MessageSecurityException(SR.Format(SR.CustomCryptoAlgorithmIsNotValidHashAlgorithm, digestMethod)));
+            }
+
+            switch (digestMethod)
+            {
+                case SecurityAlgorithms.Sha1Digest:
+                    //if (SecurityUtilsEx.RequiresFipsCompliance)
+                    //    return new SHA1CryptoServiceProvider();
+                    //else
+                        return new SHA1Managed();
+                case SecurityAlgorithms.Sha256Digest:
```

Comment:
Use SHA1.Create() and on the one below use SHA256.Create(). Remove the commented code.